File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        
        int left = 1;
        int right = numbers.length;
    
        while(numbers[left-1] + numbers[right-1] != target){         
            
            if (numbers[left-1] + numbers[right-1] > target){
                right--;
            } else {
                left++;
            }            
        }    
        return new int[]{left, right};
    }
}

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:

        left = 0
        right = len(numbers)-1

        while left < right:
            currSum = numbers[left] + numbers[right]

            if currSum == target:
                return [left+1, right+1]

            if currSum < target:
                left += 1
            else:
                right -= 1